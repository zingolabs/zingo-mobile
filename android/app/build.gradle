apply plugin: "com.android.application"
apply plugin: "com.facebook.react"
apply plugin: 'kotlin-android'

/**
 * This is the configuration block to customize your React Native Android app.
 * By default you don't need to apply any configuration, just uncomment the lines you need.
 */

react {
    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    //   The folder where the react-native Codegen package is. Default is ../node_modules/@react-native/codegen
    // codegenDir = file("../node_modules/@react-native/codegen")
    //   The cli.js file which is the React Native CLI entrypoint. Default is ../node_modules/react-native/cli.js
    // cliFile = file("../node_modules/react-native/cli.js")
    /* Variants */
    //   The list of variants to that are debuggable. For those we're going to
    //   skip the bundling of the JS bundle and the assets. By default is just 'debug'.
    //   If you add flavors like lite, prod, etc. you'll have to list your debuggableVariants.
    // debuggableVariants = ["liteDebug", "prodDebug"]
    /* Bundling */
    //   A list containing the node command and its flags. Default is just 'node'.
    // nodeExecutableAndArgs = ["node"]
    //
    //   The command to run when bundling. By default is 'bundle'
    // bundleCommand = "ram-bundle"
    //
    //   The path to the CLI configuration file. Default is empty.
    // bundleConfig = file(../rn-cli.config.js)
    //
    //   The name of the generated asset file containing your JS bundle
    // bundleAssetName = "MyApplication.android.bundle"
    //
    //   The entry file for bundle generation. Default is 'index.android.js' or 'index.js'
    // entryFile = file("../js/MyApplication.android.js")
    //
    //   A list of extra flags to pass to the 'bundle' commands.
    //   See https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle
    // extraPackagerArgs = []
    /* Hermes Commands */
    //   The hermes compiler command to run. By default it is 'hermesc'
    // hermesCommand = "$rootDir/my-custom-hermesc/bin/hermesc"
    //
    //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
    // hermesFlags = ["-O", "-output-source-map"]
}

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US.  Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

/**
 * Whether to enable building a separate APK for each ABI.
 *
 * Defaults to false but can be set to true with the project properties flag
 * e.g. ./gradlew assembleDebug -PsplitApk=true
 *
 * Additional option to include a universal APK
 * e.g. ./gradlew assembleRelease -PsplitApk=true -PincludeUniversalApk=true
 */
ext.splitApk = project.properties['splitApk'] ?: false
ext.includeUniversalApk = project.properties['includeUniversalApk'] ?: false

android {
    namespace 'org.ZingoLabs.Zingo'
    ndkVersion rootProject.ext.ndkVersion

    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility = "18.0.2.1"
        targetCompatibility = "18.0.2.1"
    }

    kotlinOptions {
        jvmTarget = '18'
    }

    defaultConfig {
        applicationId 'org.ZingoLabs.Zingo' // Real
        //applicationId 'com.ZingoMobile' // @Test
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 126 // Real
        //versionCode 117 // @Test
        versionName "zingo-1.2.5" // Real
        missingDimensionStrategy 'react-native-camera', 'general'
        testBuildType System.getProperty('testBuildType', 'debug')
        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
    }

    splits {
        abi {
            enable splitApk.toBoolean()
            reset()
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
            universalApk includeUniversalApk.toBoolean()
        }
    }

    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }

    packagingOptions {
        jniLibs {
            pickFirsts += ['lib/armeabi-v7a/libc++_shared.so', 'lib/arm64-v8a/libc++_shared.so', 'lib/x86/libc++_shared.so', 'lib/x86_64/libc++_shared.so']
        }
    }

    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            proguardFile "${rootProject.projectDir}/../node_modules/detox/android/detox/proguard-rules-app.pro"
        }
    }

    buildFeatures {
        viewBinding = true
    }

    testOptions {
        // Managed devices for gradle integration testing
        managedDevices {
            devices {
                pixel2api29_x86 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 29
                    systemImageSource = "aosp"
                    require64Bit = false
                }
                pixel2api30_x86_64 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 30
                    systemImageSource = "aosp"
                    require64Bit = true
                }
            }
            groups {
                x86_Archs {
                    targetDevices.add(devices.pixel2api29_x86)
                    targetDevices.add(devices.pixel2api30_x86_64)
                }
            }
        }
    }
    
    // // applicationVariants are e.g. debug, release
    // applicationVariants.all { variant ->
    //     variant.outputs.each { output ->
    //         // For each separate APK per architecture, set a unique version code as described here:
    //         // https://developer.android.com/studio/build/configure-apk-splits.html
    //         // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
    //         def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
    //         def abi = output.getFilter(OutputFile.ABI)
    //         if (abi != null) {  // null for the universal-debug, universal-release variants
    //             output.versionCodeOverride =
    //                     defaultConfig.versionCode * 1000 + versionCodes.get(abi)
    //         }
    //     }
    // }
}


// Map for the version code that gives each ABI a value.
ext.abiCodes = ['armeabi-v7a':1, 'x86':2, 'arm64-v8a': 3, 'x86_64':4]

import com.android.build.OutputFile

// For each APK output variant, override versionCode with a combination of
// ext.abiCodes * 10000 + variant.versionCode. 
// variant.versionCode is equal to defaultConfig.versionCode.
// If you configure product flavors that define their own versionCode, 
// variant.versionCode uses that value instead.
android.applicationVariants.all { variant ->
  variant.outputs.each { output ->
    def baseAbiVersionCode = project.ext.abiCodes.get(output.getFilter(OutputFile.ABI))

    // Because abiCodes.get() returns null for ABIs that are not mapped by ext.abiCodes,
    // the following code doesn't override the version code for universal APKs.
    // However, because you want universal APKs to have the lowest version code,
    // this outcome is desirable.
    if (baseAbiVersionCode != null) {

      // Assigns the new version code to versionCodeOverride, which changes the
      // version code for only the output APK, not for the variant itself. Skipping
      // this step causes Gradle to use the value of variant.versionCode for the APK.
      output.versionCodeOverride = baseAbiVersionCode * 10000 + variant.versionCode
    }
  }
}
            
dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    androidTestImplementation('com.wix:detox:+')
    implementation 'androidx.appcompat:appcompat:1.1.0'
    implementation fileTree(dir: "libs", include: ["*.jar"])

    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
    implementation 'com.facebook.soloader:soloader:0.10.5+'

    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }

    // Detox tests getAttributes() needs this
    debugImplementation('com.google.android.material:material:1.2.1')

    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

    if (hermesEnabled) {
        implementation("com.facebook.react:hermes-engine:+") {
            exclude group:'com.facebook.fbjni'
        }
    } else {
        implementation jscFlavor
    }
	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"

    def work_version = "2.7.1"

    // (Java only)
    implementation "androidx.work:work-runtime:$work_version"

    // Kotlin + coroutines
    implementation "androidx.work:work-runtime-ktx:$work_version"

    // optional - RxJava2 support
    implementation "androidx.work:work-rxjava2:$work_version"

    // optional - GCMNetworkManager support
    implementation "androidx.work:work-gcm:$work_version"

    // optional - Test helpers
    androidTestImplementation "androidx.work:work-testing:$work_version"

    // optional - Multiprocess support
    implementation "androidx.work:work-multiprocess:$work_version"
    
    // google truth testing framework
    androidTestImplementation "com.google.truth:truth:1.1.3"
    
    // JSON parsing
    implementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.14.+"
    
    // JUnit test runners
    androidTestImplementation 'junit:junit:4.13'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.implementation
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
repositories {
    mavenCentral()
    google()
}
