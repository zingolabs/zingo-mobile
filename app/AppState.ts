// new class with all the information about the sync process
export class SyncStatusReport {
  // sync ID
  syncID: number;

  // total of batches to process (usually 1000 blocks per batch)
  totalBatches: number;

  // batch that is processing now
  currentBatch: number;

  // total blocks per batch
  blocksPerBatch: number;

  // last block of the wallet
  lastBlockWallet: number;

  // block that is processing
  currentBlock: number;

  inProgress: boolean;

  lastError: string;

  secondsPerBatch: number;

  percent: number;

  message: string;

  process_end_block: number;

  lastBlockServer: number;

  constructor() {
    this.syncID = 0;
    this.totalBatches = 0;
    this.currentBatch = 0;
    this.blocksPerBatch = 1000; // this is set in zingolib
    this.lastBlockWallet = 0;
    this.currentBlock = 0;
    this.inProgress = false;
    this.lastError = '';
    this.secondsPerBatch = 0;
    this.percent = 0;
    this.message = '';
    this.process_end_block = 0;
    this.lastBlockServer = 0;
  }
}

export class TotalBalance {
  // Total t address, confirmed and spendable
  transparentBal: number;

  // Total private, confirmed + unconfirmed
  privateBal: number;

  // Total orchard, confirmed + unconfirmed
  orchardBal: number;

  // Total private, confirmed funds that are spendable
  spendablePrivate: number;

  // Total orchard, confirmed funds that are spendable
  spendableOrchard: number;

  // Total unconfirmed + spendable
  total: number;

  constructor() {
    this.transparentBal = 0;
    this.privateBal = 0;
    this.spendablePrivate = 0;
    this.orchardBal = 0;
    this.spendableOrchard = 0;
    this.total = 0;
  }
}

export class Address {
  uaAddress: string;
  address: string;
  addressKind: string;
  containsPending: boolean;

  constructor(uaAddress: string, address: string, addressKind: string) {
    this.uaAddress = uaAddress;
    this.address = address;
    this.addressKind = addressKind;
    this.containsPending = false;
  }
}

export class AddressBookEntry {
  label: string;

  address: string;

  constructor(label: string, address: string) {
    this.label = label;
    this.address = address;
  }
}

export class SettingsFileEntry {
  server?: string;

  constructor(server: string) {
    this.server = server;
  }
}

export interface TxDetailType {
  address: string;
  amount: number;
  memo: string | null;
}

// List of transactions.
export interface Transaction {
  type: string;
  address: string;
  amount: number;
  position: string;
  confirmations: number;
  txid: string;
  time: number;
  zec_price: number | null;
  detailedTxns: TxDetailType[];
}

// This is the type that the RPC interface expects for sending
export interface SendJsonToType {
  address: string;
  amount: number;
  memo?: string;
}

export class ToAddr {
  id: number;
  to: string;
  amount: string;
  amountUSD: string;
  memo: string;

  constructor(id: number) {
    this.id = id;

    this.to = '';
    this.amount = '';
    this.amountUSD = '';
    this.memo = '';
  }
}

export class SendPageState {
  toaddrs: ToAddr[];

  constructor() {
    this.toaddrs = [];
  }
}

export class ReceivePageState {
  // A newly created address to show by default
  newAddress: string;

  // The key used for the receive page component.
  // Increment to force re-render
  rerenderKey: number;

  constructor() {
    this.newAddress = '';
    this.rerenderKey = 0;
  }
}

export interface InfoType {
  chain_name: string;
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  zecPrice: number | null;
  defaultFee: number;
}

export class SendProgress {
  sendInProgress: boolean;
  progress: number;
  total: number;
  etaSeconds: number | string;

  constructor() {
    this.sendInProgress = false;
    this.progress = 0;
    this.total = 0;
    this.etaSeconds = 0;
  }
}

export interface ServerSelectState {
  modalIsOpen: boolean;
}

export class PasswordState {
  showPassword: boolean;

  confirmNeeded: boolean;

  passwordCallback: ((password: string) => void) | null;

  closeCallback: (() => void) | null;

  helpText: string | null;

  constructor() {
    this.showPassword = false;
    this.confirmNeeded = false;
    this.passwordCallback = null;
    this.closeCallback = null;
    this.helpText = null;
  }
}

export class ErrorModalData {
  title: string;
  body: string;

  modalIsOpen: boolean;

  constructor() {
    this.modalIsOpen = false;
    this.title = '';
    this.body = '';
  }
}

export interface ConnectedCompanionApp {
  name: string;
  lastSeen: number;
}

export interface WalletSeed {
  seed: string;
  birthday: number;
}

export interface SyncStatus {
  inProgress: boolean;
  progress: number;
  blocks: string;
}

export class WalletSettings {
  download_memos: string;
  transaction_filter_threshold: string;
  server?: string;

  constructor() {
    this.download_memos = 'wallet';
    this.transaction_filter_threshold = '500';
  }
}

export default interface AppState {
  // Info about the current sync process
  syncStatusReport: SyncStatusReport;

  // The total confirmed and unconfirmed balance in this wallet
  totalBalance: TotalBalance;

  // A map type that contains address -> privatekey mapping, for display on the receive page
  // This mapping is ephemeral, and will disappear when the user navigates away.
  addressPrivateKeys: Map<string, string>;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: Address[];

  // List of Address / Label pairs
  addressBook: AddressBookEntry[];

  // List of all T and Z and O transactions
  transactions: Transaction[] | null;

  // The state of the send page, as the user constructs a transaction
  sendPageState: SendPageState;

  // Any state for the receive page
  receivePageState: ReceivePageState;

  // getinfo and getblockchaininfo result
  info: InfoType | null;

  // Is the app rescanning?
  rescanning: boolean;

  // Callbacks for the password dialog box
  //passwordState: PasswordState;

  wallet_settings: WalletSettings;

  syncingStatus: SyncStatus | null;

  // Data for any error or info modal
  errorModalData: ErrorModalData;

  // Build progress from Tx
  txBuildProgress: SendProgress;

  walletSeed: WalletSeed | null;

  isMenuDrawerOpen: boolean;

  selectedMenuDrawerItem: string;

  aboutModalVisible: boolean;

  computingModalVisible: boolean;

  settingsModalVisible: boolean;

  infoModalVisible: boolean;

  rescanModalVisible: boolean;

  seedViewModalVisible: boolean;
  seedChangeModalVisible: boolean;
  seedBackupModalVisible: boolean;
  seedServerModalVisible: boolean;

  syncReportModalVisible: boolean;

  newServer: string | null;

  uaAddress: string | null;

  // eslint-disable-next-line semi
}
