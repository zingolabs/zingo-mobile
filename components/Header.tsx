import { ReactNode } from "react"
import { View } from "react-native"
import { useTheme } from '@react-navigation/native'
import { ThemeType } from "../app/types/ThemeType"
import ZingoLogoImage from "./ZingoLogoImage"

type HeaderProps = {
    children: ReactNode
}


const Header: React.FunctionComponent<HeaderProps> = (props) => {
    // TODO: why?
    const { colors } = useTheme() as unknown as ThemeType
    console.log(colors)

    console.log("THEME", useTheme())

    return (
        <View
            style={{
                display: 'flex',
                alignItems: 'center',
                // paddingBottom: 10,
                // backgroundColor: colors.card ||,
                // zIndex: -1,
                // paddingTop: 10,
            }}>
            <ZingoLogoImage />
            {props.children}
            <View style={{ width: '100%', height: 1, backgroundColor: colors.primary }} />
        </View>
    )
}

export default Header
