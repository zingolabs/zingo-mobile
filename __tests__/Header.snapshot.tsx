/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Header from '../components/Header';
import { defaultAppStateLoaded, ContextAppLoadedProvider } from '../app/context';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component Header - test', () => {
  //snapshot test
  test('Header Simple - snapshot', () => {
    const state = defaultAppStateLoaded;
    state.translate = () => 'text translated';
    const about = render(
      <ContextAppLoadedProvider value={state}>
        <Header title="title" noBalance={true} noSyncingStatus={true} noDrawMenu={true} noPrivacy={true} />
      </ContextAppLoadedProvider>,
    );
    expect(about.toJSON()).toMatchSnapshot();
  });
  test('Header Complex - snapshot', () => {
    const state = defaultAppStateLoaded;
    state.translate = () => 'text translated';
    state.info.currencyName = 'ZEC';
    state.totalBalance.total = 1.12345678;
    const onFunction = jest.fn();
    const header = render(
      <ContextAppLoadedProvider value={state}>
        <Header
          testID="transaction text"
          toggleMenuDrawer={onFunction}
          setZecPrice={onFunction}
          title="title"
          setComputingModalVisible={onFunction}
          setBackgroundError={onFunction}
          set_privacy_option={onFunction}
          setPoolsToShieldSelectSapling={onFunction}
          setPoolsToShieldSelectTransparent={onFunction}
          setUfvkViewModalVisible={onFunction}
          addLastSnackbar={onFunction}
        />
      </ContextAppLoadedProvider>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
