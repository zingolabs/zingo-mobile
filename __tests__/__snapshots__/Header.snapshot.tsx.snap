// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header - test Header Complex - snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255, 255, 255)",
      "display": "flex",
      "paddingBottom": 0,
      "paddingTop": 10,
      "zIndex": -1,
    }
  }
  testID="header"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 40,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": undefined,
          "borderRadius": 10,
          "borderWidth": 1,
          "justifyContent": "center",
          "margin": 0,
          "marginRight": 5,
          "minHeight": 25,
          "minWidth": 25,
          "padding": 1,
        }
      }
    >
      <View
        style={
          Object {
            "margin": 0,
            "padding": 0,
          }
        }
      >
        <
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [
                  "wifi-3",
                  "wifi-strong",
                ],
                "f1eb",
                "M54.2 202.9C123.2 136.7 216.8 96 320 96s196.8 40.7 265.8 106.9c12.8 12.2 33 11.8 45.2-.9s11.8-33-.9-45.2C549.7 79.5 440.4 32 320 32S90.3 79.5 9.8 156.7C-2.9 169-3.3 189.2 8.9 202s32.5 13.2 45.2 .9zM320 256c56.8 0 108.6 21.1 148.2 56c13.3 11.7 33.5 10.4 45.2-2.8s10.4-33.5-2.8-45.2C459.8 219.2 393 192 320 192s-139.8 27.2-190.5 72c-13.3 11.7-14.5 31.9-2.8 45.2s31.9 14.5 45.2 2.8c39.5-34.9 91.3-56 148.2-56zm64 160a64 64 0 1 0 -128 0 64 64 0 1 0 128 0z",
              ],
              "iconName": "wifi",
              "prefix": "fas",
            }
          }
          size={18}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 0,
        "marginTop": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "margin": 5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="25.2"
            bbWidth="50.4"
            fill="rgb(28, 28, 30)"
            focusable={false}
            height={25.2}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "marginBottom": 6,
                },
                Object {
                  "flex": 0,
                  "height": 25,
                  "width": 50,
                },
              ]
            }
            vbHeight={147.85}
            vbWidth={282.84}
            width={50.4}
            xml="<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
  <svg viewBox=\\"0 0 282.84 147.85\\">
    <polygon points=\\"34.44 107.62 34.44 106.98 87.19 34.17 87.19 20.12 56.09 20.12 56.09 0 35.98 0 35.98 20.12 5.04 20.12 5.04 40.24 53.93 40.24 53.93 40.88 0 114.64 0 127.73 35.98 127.73 35.98 147.85 56.09 147.85 56.09 127.73 88.03 127.73 88.03 107.62 34.44 107.62\\"/>
    <polygon points=\\"180.89 39.03 180.89 19.35 124.56 19.35 110.48 19.35 102.52 19.35 102.52 127.75 110.48 127.75 124.56 127.75 180.89 127.75 180.89 108.07 124.56 108.07 124.56 83.77 172.04 83.77 172.04 64.09 124.56 64.09 124.56 39.03 180.89 39.03\\"/>
    <path d=\\"m240.92,128.87c-8.96,0-16.59-1.58-22.89-4.73-6.3-3.16-11.12-7.75-14.44-13.79-3.33-6.04-4.99-13.33-4.99-21.88v-29.75c0-8.6,1.66-15.9,4.99-21.92,3.33-6.01,8.14-10.61,14.44-13.79,6.3-3.18,13.93-4.77,22.89-4.77,7.4,0,13.99,1.47,19.75,4.4,5.77,2.93,10.54,7.18,14.32,12.75,3.78,5.57,6.4,12.33,7.85,20.28h-22.85c-.86-3.63-2.19-6.7-3.98-9.21-1.8-2.51-3.97-4.44-6.52-5.78-2.55-1.34-5.41-2.01-8.57-2.01-6.17,0-10.96,1.75-14.36,5.26-3.41,3.5-5.11,8.44-5.11,14.8v29.75c0,6.36,1.7,11.28,5.11,14.76,3.41,3.48,8.19,5.22,14.36,5.22,4.83,0,8.9-1.48,12.23-4.44,3.33-2.96,5.61-7.14,6.84-12.56h22.85c-1.5,7.9-4.14,14.65-7.93,20.24-3.78,5.59-8.54,9.85-14.28,12.79-5.74,2.93-12.31,4.4-19.71,4.4Z\\"/>
  </svg>"
          >
            <RNSVGGroup
              fill={
                Object {
                  "payload": 4280032286,
                  "type": 0,
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M34.44 107.62 34.44 106.98 87.19 34.17 87.19 20.12 56.09 20.12 56.09 0 35.98 0 35.98 20.12 5.04 20.12 5.04 40.24 53.93 40.24 53.93 40.88 0 114.64 0 127.73 35.98 127.73 35.98 147.85 56.09 147.85 56.09 127.73 88.03 127.73 88.03 107.62 34.44 107.62z"
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M180.89 39.03 180.89 19.35 124.56 19.35 110.48 19.35 102.52 19.35 102.52 127.75 110.48 127.75 124.56 127.75 180.89 127.75 180.89 108.07 124.56 108.07 124.56 83.77 172.04 83.77 172.04 64.09 124.56 64.09 124.56 39.03 180.89 39.03z"
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="m240.92,128.87c-8.96,0-16.59-1.58-22.89-4.73-6.3-3.16-11.12-7.75-14.44-13.79-3.33-6.04-4.99-13.33-4.99-21.88v-29.75c0-8.6,1.66-15.9,4.99-21.92,3.33-6.01,8.14-10.61,14.44-13.79,6.3-3.18,13.93-4.77,22.89-4.77,7.4,0,13.99,1.47,19.75,4.4,5.77,2.93,10.54,7.18,14.32,12.75,3.78,5.57,6.4,12.33,7.85,20.28h-22.85c-.86-3.63-2.19-6.7-3.98-9.21-1.8-2.51-3.97-4.44-6.52-5.78-2.55-1.34-5.41-2.01-8.57-2.01-6.17,0-10.96,1.75-14.36,5.26-3.41,3.5-5.11,8.44-5.11,14.8v29.75c0,6.36,1.7,11.28,5.11,14.76,3.41,3.48,8.19,5.22,14.36,5.22,4.83,0,8.9-1.48,12.23-4.44,3.33-2.96,5.61-7.14,6.84-12.56h22.85c-1.5,7.9-4.14,14.65-7.93,20.24-3.78,5.59-8.54,9.85-14.28,12.79-5.74,2.93-12.31,4.4-19.71,4.4Z"
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 36,
                "fontWeight": "700",
                "margin": 0,
                "padding": 0,
              }
            }
            testID="undefined.big-part"
          >
             1.1234
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 25.2,
                "margin": 0,
                "marginBottom": 2.4,
                "padding": 0,
              }
            }
            testID="undefined.small-part"
          >
            5678
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(28, 28, 30)",
          "fontSize": 18,
          "fontWeight": "600",
          "opacity": 1,
          "paddingHorizontal": 5,
        }
      }
      testID="transaction text"
    >
      title
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "left": 0,
        "padding": 11.5,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="text translated"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 5,
            "opacity": 1,
          }
        }
        testID="header.drawmenu"
      >
        <
          color="rgb(216, 216, 216)"
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [
                  "navicon",
                ],
                "f0c9",
                "M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z",
              ],
              "iconName": "bars",
              "prefix": "fas",
            }
          }
          size={45}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "padding": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/logobig-zingo.png",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 38,
          "resizeMode": "contain",
          "width": 38,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 255)",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Component Header - test Header Simple - snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255, 255, 255)",
      "display": "flex",
      "paddingBottom": 0,
      "paddingTop": 10,
      "zIndex": -1,
    }
  }
  testID="header"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 40,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "marginTop": 12,
      }
    }
  />
  <View
    style={
      Object {
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(28, 28, 30)",
          "fontSize": 18,
          "fontWeight": "600",
          "opacity": 1,
          "paddingHorizontal": 5,
        }
      }
    >
      title
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "left": 0,
        "padding": 11.5,
        "position": "absolute",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "padding": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/logobig-zingo.png",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 38,
          "resizeMode": "contain",
          "width": 38,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 255)",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;
